File: articles.jdbc/src/test/java/pet/db/jdbc/tool/generator/ReviewPayloadTestDataGenerator.java
package pet.db.jdbc.tool.generator;

import lombok.RequiredArgsConstructor;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import pet.db.jdbc.model.dto.payload.ReviewPayload;
import pet.db.jdbc.model.dto.Review;

import java.util.List;

@Component
@RequiredArgsConstructor
public class ReviewPayloadTestDataGenerator implements TestDataGenerator<ReviewPayload> {

    @Autowired
    private TestDataGenerator<Review> reviewTestDataGenerator;

    @Override
    public ReviewPayload generateSavedData() {
        return generateSavedData(1).getFirst();
    }

    @Override
    public ReviewPayload generateUnsavedData() {
        return generateUnsavedData(1).getFirst();
    }

    @Override
    public List<ReviewPayload> generateSavedData(Integer dataSize) {
        return reviewTestDataGenerator.generateSavedData(dataSize).stream()
                .map(this::convertToReviewPayload)
                .toList();
    }

    @Override
    public List<ReviewPayload> generateUnsavedData(Integer dataSize) {
        return reviewTestDataGenerator.generateUnsavedData(dataSize).stream()
                .map(this::convertToReviewPayload)
                .toList();
    }

    private ReviewPayload convertToReviewPayload(Review review) {
        return new ReviewPayload(
                review.getType(),
                review.getContent(),
                review.getAuthorId(),
                review.getArticleId());
    }

}

File: src/test/kotlin/pet/articles/tool/generator/ReviewPayloadTestDataGenerator.kt
package pet.articles.tool.generator

import org.koin.core.annotation.Named
import org.koin.core.annotation.Single

import pet.articles.model.dto.Review
import pet.articles.model.dto.payload.ReviewPayload

class ReviewPayloadTestDataGenerator(
    private val reviewGenerator: TestDataGenerator<Review>
) : TestDataGenerator<ReviewPayload> {

    override fun generateUnsavedData(dataSize: Int): List<ReviewPayload> =
        reviewGenerator.generateUnsavedData(dataSize).map(::convertToReviewPayload)

    override fun generateSavedData(dataSize: Int): List<ReviewPayload> =
        reviewGenerator.generateSavedData(dataSize).map(::convertToReviewPayload)

    override fun generateInvalidData(): ReviewPayload =
        convertToReviewPayload(reviewGenerator.generateInvalidData())

    private fun convertToReviewPayload(review: Review): ReviewPayload =
        ReviewPayload(
            type = review.type,
            content = review.content,
            authorId = review.authorId!!,
            articleId = review.articleId
        )
}